Development/OIDC.md

```diff
@@ -1,169 +1,34 @@
-        "authPath": "/auth/google",
-        "successRedirect": "http://localhost/user",
-        "failureRedirect": "http://localhost/login",
-        "failureFlash": true,
-        "session": false,
-        "issuer": "https://accounts.google.com",
-        "authorizationURL": "https://accounts.google.com/o/oauth2/v2/auth",
-        "tokenURL": "https://oauth2.googleapis.com/token",
-        "userInfoURL": "https://openidconnect.googleapis.com/v1/userinfo",
-        "clientID": "...",
-        "clientSecret": "...",
-        "callbackURL": "http://localhost:3000/auth/google/callback",
-        "scope": ["email", "profile", "openid"],
-        "loginCallback": "sampleLoginCallback"
-    }
-```
-> Note that information about Google sign in can be found [here](https://developers.google.com/identity/protocols/oauth2/openid-connect), details are beyond the scope of this document.
-
-
-> Note that OIDC authentication services publish their specific settings in a Discovery document.. To find the settings for, say, google, see <https://accounts.google.com/.well-known/openid-configuration>.
-
-Details about the fields in this provider configuration:
-* **provider** is used by the loopback-passport-configurator to build a strategy
-* **authScheme** is used by the loopback-passport-configurator to build a strategy
-* **module** describes the passport module to use. passport-openidconnect is already included in the backend
-* **authPath** is the path that the backend will create (with the help of loopback-passport-configurator) to begin the authentication process. The backend will redirect users to this path (once it is configured, of course!)
-* **successRedirect** is the URL passed to the OIDC provider instructing it to redirect the user on success. Here we set it to the a development environment's user page.
-* **failureRedirect** is the URL passed to the OIDC provider instructing it to redirect the user on failed authentication. Here we set it to the a development environment's user page login page.
-* **failureFlash** is currently unsupported.
-* **session** is not needed, since the frontend authenticates outside of sessions.
-* **authorizationURL** is specific to your OIDC provider
-* **tokenURL** is specific to your OIDC provider
-* **userInfoURL** is specific to your OIDC provider
-* **clientID** is specific to your OIDC provider
-* **clientSecret** is specific to your OIDC provider
-* **scope** is specific to your OIDC provider
-* **loginCallback** optional name of a [callback](#configure-callback) function to run to add profile information to the UserIdentity
-
-#### Session Secret 
-The nodejs module `passport-openidconnect` requires the use of the `express-session` module
-to create a login session for the user.
- This requires a _session secret_ to be configured in `config.local.js`:
- ```
- module.exports = {
-  expressSessionSecret: "thisIsSuperSecretForUserSession"
- ```
-
-## SciCat Frontend Configuration 
```

Comment:
I think this front end configuration stays the same, and should not be deleted.

Reply:
thanks for pointing this out - I have put it back.